src/tiled/automapper.cpp

```diff
@@ -712,8 +712,18 @@ static bool compareLayerTo(const TileLayer *setLayer,
                 bool matchListYes = false;
                 bool matchListNo  = false;
 
-                if (!setLayer->contains(x + offset.x(), y + offset.y()))
-                    return false;
+
+                if (!setLayer->contains(x + offset.x(), y + offset.y())) {
+                    foreach (const TileLayer *comparedTileLayer, listYes) {
+                        if (!comparedTileLayer->contains(x, y))
+                            return false;
```

Comment:
> which comes from 'the stone may be just outside the map', and we don't want to have that IMHO.

Right, but I was asking why the match failed when a layer in `listYes` does not contain this rule region position, whereas in the case of the stone, the `listYes` does contain the position but the `setLayer` doesn't, right?